### DISCUSSION TITLE
How do I install libtensorflow for the Recognize app to run natively?

### DISCUSSION THREAD

**Discussion by pepper-lees** [nextcloud/all-in-one#2330] [Questions] [answered]
Hi,

The Recognize app expects libtensorflow to be installed in order to run natively, but it seems that that would require modifying the nextcloud-aio-nextcloud container. I got instructions from ChatGPT to clone the nextcloud AIO image and install libtensorflow in it, but I don't think it'll work because libtensorflow will be added to the master container, not the nextcloud container - Right?

```Dockerfile
FROM nextcloud/all-in-one:latest

# Install dependencies for libtensorflow
RUN apt-get update \
    && apt-get install -y libtensorflow1

```
I was instructed to then run `docker build -t nextcloud-with-tensorflow .` and then replace `nextcloud/all-in-one:latest` in my `docker run command with `nextcloud-with-tensorflow`.

I'm hesitant to test it because I'm really not familiar with Docker and I'm afraid of messing something up. I'd rather ask for assistance. Would this technique work, or would libtensorflow only be installed in the mastercontainer, leading to Recognize not having access to it? If it's the latter, then how do I do it properly?

**Comment by szaimen** [comment, marked as answer]
Hi, libtensorflow does unfortunately not work with native speed inside an Alpine based container.

However  I would gladly add an additional container for this after https://github.com/nextcloud/recognize/issues/73 is implemented.

**Reply by Zoey2936** [reply]
just for me, why does it not run with full speed?

**Reply by szaimen** [reply]
Because it needs glibc for that.

**Reply by Zoey2936** [reply]
can you maybe give me links where this is written, please?

**Reply by pepper-lees** [reply]
@szaimen Correct me if I'm wrong but I think it's simply because the AIO docker image doesn't contain the necessary libraries. If someone really wanted to, they could fork the image and make their own version with the necessary dependencies, right?

**Reply by szaimen** [reply]
tensorflow is simply incompatible with alpine linux. See https://github.com/nextcloud/all-in-one/pull/1569#issuecomment-1399040724 and https://github.com/tensorflow/tfjs/issues/1425. Because of that only WASM mode is poasible
